### ISSUE TITLE
lua.decode('"--3"') fails

### ISSUE THREAD

**Issue by larsimmisch** [SirAnthony/slpp#23]
This looks like a bug in the string parsing to me - as if the parser tries to detect comments inside literal strings.

`'--3'` is a perfectly valid lua string, of course.

**Comment by larsimmisch** [comment]
This is actually the FIXME in line 33.

**Comment by Dimbreath** [comment]
Any fix for this?

**Comment by lantisnt** [comment]
@SirAnthony bumping. Would love to get this fixed. If you dont have time i can do a pull request. Only some testing and releasing then to pip would be lovely

**Comment by SirAnthony** [comment]
> @SirAnthony bumping. Would love to get this fixed. If you dont have time i can do a pull request. Only some testing and releasing then to pip would be lovely

Sure, pull requests are welcome. Please bring some tests for that also, it is 3-lines change in tests.py, but helps a lot.

**Comment by harmtemolder** [comment]
I just ran into this as well. Changing the regex to the following only removes comments when they are the first thing on a line (may be preceded by whitespace), which fixed the issue I was having:

```python
reg = re.compile('^\s*--.*$', re.M)
```

This does however leave two issues:

1. It still removes lines in multiline strings where a line starts with `--`:

    ```lua
    ["multiline_string"] = "A multiline string where one of the lines starts with
    -- two dashes"
    ```

    (Results in a parse error because the end quote is missing)

    ```lua
    ["another_multiline_string"] = "A multiline string where one of the lines starts with
    -- two dashes,
    followed by another line"
    ```

    (Results in the line being dropped from the string)

2. It does not remove trailing comments:

    ```lua
    ["trailing_comment"] = "A string with" -- a trailing comment
    ```

I believe the issue of removing comments is too complex to be handled by this regex. I’ll have a look at how the loops work and where the handling of comments should be integrated.

**Comment by SirAnthony** [comment]
Fixed in #32, thanks. I also added special test for this issue case, and support for simple multiline comments.
